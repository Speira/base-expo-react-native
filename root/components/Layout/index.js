import PropTypes from 'prop-types'
import BaseLayout, {
  LayoutBody,
  LayoutFooter,
  LayoutHeader,
  LayoutStatusBar,
  LayoutTitle,
} from './style'

/**
 * Layout
 * @desc ::: create a layout for the qpp
 *
 */
export default function Layout(props) {
  const { children, title } = props

  return (
    <BaseLayout>
      <LayoutStatusBar />
      <LayoutHeader>
        <LayoutTitle>{title}</LayoutTitle>
      </LayoutHeader>
      <LayoutBody>{children}</LayoutBody>
      <LayoutFooter />
    </BaseLayout>
  )
}

Layout.defaultProps = {
  title: '',
}
Layout.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string,
}
