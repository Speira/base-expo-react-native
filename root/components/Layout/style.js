import styled from 'styled-components/native'
import { SafeAreaView as View } from 'react-native-safe-area-context'
import { StatusBar, Text } from 'react-native'

/**
 * LayoutStatusBar
 * @component
 *
 */
export const LayoutStatusBar = styled(StatusBar).attrs(({ theme }) => ({
  barStyle: 'default',
  backgroundColor: theme.COLORS.STATIC.DARK,
}))``

/**
 * LayoutTitle
 * @component
 *
 */
export const LayoutTitle = styled(Text)`
  font-size: 32px;
  font-weight: 600;
  text-align: center;
  color: ${({ theme }) => theme.COLORS.STATIC.DARK};
`
/**
 * LayoutHeader
 * @component
 *
 */
export const LayoutHeader = styled(View)`
  flex: 0.12;
`
/**
 * LayoutBody
 * @component
 *
 */
export const LayoutBody = styled(View)`
  flex: 0.76;
  justify-content: space-between;
`
/**
 * LayoutFooter
 * @component
 *
 */
export const LayoutFooter = styled(View)`
  flex: 0.1;
`

/**
 * BaseLayout
 * @component
 *
 */
const BaseLayout = styled(View)`
  flex: 1;
  background-color: ${({ theme }) => theme.COLORS.STATIC.PRIMARY};
  padding-top: 10px;
`

export default BaseLayout
