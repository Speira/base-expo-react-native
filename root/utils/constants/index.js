// utils/constants
// description: We try to set all constants into a big one to reduce Javascript
//              constants performance issue.

export default {
  API: {
    FETCH_DATA_URL:
      'https://static.leboncoin.fr/img/shared/technical-test.json',
    PROXY: {
      NO_CORS_SERVER: 'https://cors-anywhere.herokuapp.com/',
    },
    TIMEOUT: 20000,
  },
  LABELS: {
    CLEAR_DATABASE: 'Clear the database',
    START_DATABASE: 'Create the database',
    IMVALID_DATA_FORMAT: ' Invalid data format',
    LIST: 'Item List',
    NO_CARD: 'No card to persist',
    NO_DATABASE: 'The database is not set',
    NO_RESPONSE_DATA: 'There is no data in the query response',
    REFETCH_DATA: 'Refetch Data',
    SEARCH_DATA_FAILED:
      'No data for SearchDataAPI: Please unlock the proxy below and then refetch data',
    WELCOME: 'Welcome to the ListItem app',
  },
  STATES: {
    ERROR: 'error',
    LOADING: 'loading',
  },
  STATUS: {
    DANGER: 'DANGER',
    INFO: 'INFO',
    SUCCESS: 'SUCCESS',
    WARNING: 'WARNING',
  },
  STORAGE_ITEMS: {
    DB: 'storage_list_item_app',
    CARD: 'card_items.db',
    THEME: 'app_theme_storage',
  },
  THEMES: {
    DEFAULT: {
      COLORS: {
        STATIC: {
          PRIMARY: '#EBD8C3',
          SECONDARY: '#F7E9D7',
          TERTIARY: '#FFF6EA',
          QUATERNARY: '#FFEEEE',
          LIGHT: '#f8f8fa',
          DARK: '#4d3e30',
        },
        DYNAMIC: {
          INFO: '#035397',
          SUCCESS: '#1C7947',
          WARNING: '#C36A2D',
          DANGER: '#A9333A',
        },
      },
    },
  },
}
