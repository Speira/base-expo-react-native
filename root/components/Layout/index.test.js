import React from 'react'
import { shallow } from 'enzyme'
import constants from '~utils/constants'
import Layout from './index'
import BaseLayout, { LayoutTitle } from './style'

const { THEMES } = constants

describe('render Layout', () => {
  const wrapper = shallow(
    <Layout title="test" theme={THEMES.DEFAULT}>
      test
    </Layout>,
  )
  it('Layout should be defined', () => {
    expect(wrapper).toBeDefined()
  })
  it('Layout should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('Layout should have a LayoutTitle', () => {
    const title = wrapper.find(LayoutTitle)
    expect(title.text()).toEqual('test')
  })
})
