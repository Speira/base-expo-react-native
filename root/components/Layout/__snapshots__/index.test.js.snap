// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Layout Layout should match snapshot 1`] = `
<style__BaseLayout>
  <style__LayoutStatusBar />
  <style__LayoutHeader>
    <style__LayoutTitle>
      test
    </style__LayoutTitle>
  </style__LayoutHeader>
  <style__LayoutBody>
    test
  </style__LayoutBody>
  <style__LayoutFooter />
</style__BaseLayout>
`;
